### PR TITLE
Add org.gmusicbrowser.gmusicbrowser

### PR DESCRIPTION
**Pull request by BeatLink**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by hfiguiere** [comment]
appstream validation fail

File: org.gmusicbrowser.gmusicbrowser.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        "--filesystem=xdg-music",
+        "--socket=pulseaudio",
+        "--socket=session-bus",
+        "--env=PERL5LIB=/app/lib"
+    ],
+    "build-options" : {
+        "env": {"PERL5LIB": "/app/lib"}
+    },
+    "modules": [
+        "audio-libs.json",
```
Comment (bilelmoussaoui) [inline org.gmusicbrowser.gmusicbrowser.json:21]: Any reason for splitting those into a new file?

File: org.gmusicbrowser.gmusicbrowser.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "sdk": "org.gnome.Sdk",
+    "command": "gmusicbrowser",
+    "finish-args": [
+        "--share=ipc", 
+        "--socket=x11",
+        "--socket=wayland",
+        "--share=network",
+        "--filesystem=xdg-music",
+        "--socket=pulseaudio",
+        "--socket=session-bus",
```
Comment (bilelmoussaoui) [inline org.gmusicbrowser.gmusicbrowser.json:14]: Does it need access to the whole session-bus?

File: org.gmusicbrowser.gmusicbrowser.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+    "app-id": "org.gmusicbrowser.gmusicbrowser",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.36",
+    "sdk": "org.gnome.Sdk",
+    "command": "gmusicbrowser",
+    "finish-args": [
+        "--share=ipc", 
+        "--socket=x11",
```
Comment (bilelmoussaoui) [inline org.gmusicbrowser.gmusicbrowser.json:9]:
```suggestion
        "--socket=fallback-x11",
```

**Comment by nedrichards** [comment]
Upstream request and context: https://github.com/squentin/gmusicbrowser/issues/173

File: org.gmusicbrowser.gmusicbrowser.json (hunk 1 above)
Comment (BeatLink) [reply org.gmusicbrowser.gmusicbrowser.json:21]: Not any particular reason, just for reusability and to make it a bit cleaner

File: org.gmusicbrowser.gmusicbrowser.json (hunk 2 above)
Comment (BeatLink) [reply org.gmusicbrowser.gmusicbrowser.json:14]: Hmm, probably not.

File: org.gmusicbrowser.gmusicbrowser.json (hunk 3 above)
Comment (BeatLink) [reply org.gmusicbrowser.gmusicbrowser.json:9]: Thank ya

**Comment by BeatLink** [comment]
Right now, im waiting on upstream to rename their appstream file. They do have the file, its just not named in the reverse dns format, i think thats why the automated tests failed on it

**Comment by hfiguiere** [comment]
> Right now, im waiting on upstream to rename their appstream file. They do have the file, its just not named in the reverse dns format, i think thats why the automated tests failed on it

You can rename it in the manifest:

https://docs.flatpak.org/en/latest/manifests.html?highlight=rename#file-renaming

**Comment by BeatLink** [comment]
ooh thanks!

File: org.gmusicbrowser.gmusicbrowser.json (hunk 2 above)
Comment (BeatLink) [reply org.gmusicbrowser.gmusicbrowser.json]: I removed it for now, Ill nail down which specific DBUS channel it needs later

**Comment by BeatLink** [comment]
I made the changes. Feel free to run the build again. Fingers crossed!

**Comment by bilelmoussaoui** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by bilelmoussaoui** [comment]
The appdata file lacks the following tags
```
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
```

**Comment by BeatLink** [comment]
Hmm, looks like ill have to contact upstream

**Comment by bilelmoussaoui** [comment]
> Hmm, looks like ill have to contact upstream

You can patch it here in the meantime, a lot of repos do that. Just make sure to submit the patch upstream if possible.

**Comment by BeatLink** [comment]
sure thing

**Comment by BeatLink** [comment]
Changes made. Give her a go!

**Comment by bilelmoussaoui** [comment]
You can do that as well
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
Any ideas why this wont work? Not sure why the other architectures are failing

**Comment by hfiguiere** [comment]
According to the logs, vorbistools don't build on aarch64 because the configure files are too old. That happens. Note that the build log talks about it. Are you sure you need to build it? libvorbis is in the SDK. So is libmpg123.

On arm, it feels like the build hard coded x86_64 in the architecture and that's were it fails. It is further along than aarch64.

File: audio-libs.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
+    "name": "audio-libs",
+    "modules": [
+        {
+            "name": "mpg123",
+            "buildsystem": "autotools",
+            "sources": [{ 
+               "type": "archive",
+                "url": "https://iweb.dl.sourceforge.net/project/mpg123/mpg123/1.26.1/mpg123-1.26.1.tar.bz2",
+                "sha256": "74d6629ab7f3dd9a588b0931528ba7ecfa989a2cad6bf53ffeef9de31b0fe032"
```
Comment (hfiguiere) [inline audio-libs.json:10]: the SDK has this:
`lrwxrwxrwx 1 nfsnobody nfsnobody     19 Jun  1 21:28 /usr/lib/x86_64-linux-gnu/libmpg123.so.0 -> libmpg123.so.0.44.9`

File: audio-libs.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                "sha256": "2f96da02c28730fcc2c71e9e6975268a4b01b0a298f1bea58a5543192f972b66"
+            }]
+        },
+        {
+            "name": "vorbis-tools",
+            "buildsystem": "autotools",
+            "sources": [{ 
+                "type": "archive",
+                "url": "http://downloads.xiph.org/releases/vorbis/vorbis-tools-1.4.0.tar.gz",
+                "sha256": "a389395baa43f8e5a796c99daf62397e435a7e73531c9f44d9084055a05d22bc"
```
Comment (hfiguiere) [inline audio-libs.json:37]: The SDK has this
```
$ ls -l /usr/lib/x86_64-linux-gnu/libvorbis*
lrwxrwxrwx 1 nfsnobody nfsnobody     22 Jun  1 21:28 /usr/lib/x86_64-linux-gnu/libvorbisenc.so.2 -> libvorbisenc.so.2.0.11
-rwxr-xr-x 5 nfsnobody nfsnobody 694448 Dec 31  1969 /usr/lib/x86_64-linux-gnu/libvorbisenc.so.2.0.11
lrwxrwxrwx 1 nfsnobody nfsnobody     22 Jun  1 21:28 /usr/lib/x86_64-linux-gnu/libvorbisfile.so.3 -> libvorbisfile.so.3.3.7
-rwxr-xr-x 5 nfsnobody nfsnobody  38944 Dec 31  1969 /usr/lib/x86_64-linux-gnu/libvorbisfile.so.3.3.7
lrwxrwxrwx 1 nfsnobody nfsnobody     18 Jun  1 21:28 /usr/lib/x86_64-linux-gnu/libvorbis.so.0 -> libvorbis.so.0.4.8
-rwxr-xr-x 5 nfsnobody nfsnobody 190568 Dec 31  1969 /usr/lib/x86_64-linux-gnu/libvorbis.so.0.4.8
```

File: perl-libs.json [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        },
+        {
+            "name": "perl-glib",
+            "buildsystem": "simple",
+            "build-commands": [
+                "perl Makefile.PL PREFIX=/app LIB=/app/lib",
+                "make",
+                "make test",
+                "make install",
+                "chmod 755 /app/lib/x86_64-linux/auto/Glib/Glib.so"
```
Comment (hfiguiere) [inline perl-libs.json:76]: This will fail on arch != x86_64. 

And below too.

File: perl-libs.json [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+        },
+        {
+            "name": "perl-glib",
+            "buildsystem": "simple",
+            "build-commands": [
+                "perl Makefile.PL PREFIX=/app LIB=/app/lib",
+                "make",
+                "make test",
+                "make install",
+                "chmod 755 /app/lib/*"
```
Comment (hfiguiere) [inline perl-libs.json:76]: Did you test this? chmod isn't recursive by default.

File: perl-libs.json [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+        },
+        {
+            "name": "perl-Glib-Object-Introspection",
+            "buildsystem": "simple",
+            "build-commands": [
+                "perl Makefile.PL PREFIX=/app LIB=/app/lib",
+                "make",
+                "make test",
+                "make install",
+                "chmod 755 /app/lib/x86_64-linux/auto/Glib/Object/Introspection/Introspection.so"
```
Comment (hfiguiere) [inline perl-libs.json:92]: Still still won't work on non x86_64. And below too.

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by hfiguiere** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by hfiguiere** [comment]
What other flatpak do is that they put in the `post-install` a `chmod -Rf u+w /app/lib/perl5/site_perl` for each perl module. This should work.

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by hfiguiere** [comment]
You should try to build locally before pushing changes that don't work. It helps a lot.

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
OH MY GOD

**Comment by BeatLink** [comment]
FINALLY!

**Comment by BeatLink** [comment]
It runs but only with dbus disabled. Otherwise it crashes with this message:

`org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown`

Any ideas why?

**Comment by BeatLink** [comment]
Its also saying 

```
These commands were not found : mpg123, ogg123, mpg321
 => these file types won't be played by the 123 output : mp3, oga
```

Thats why i installed those in the audio libs file. I thought flatpak included these

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by hfiguiere** [comment]
> It runs but only with dbus disabled. Otherwise it crashes with this message:
> 
> `org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown`
> 
> Any ideas why?

You need to add the proper `--talk-name` finish-args for DBus to be allowed.

**Comment by hfiguiere** [comment]
> Its also saying
> 
> ```
> These commands were not found : mpg123, ogg123, mpg321
>  => these file types won't be played by the 123 output : mp3, oga
> ```
> 
> Thats why i installed those in the audio libs file. I thought flatpak included these

It include the libs only. :-/

**Comment by hfiguiere** [comment]
For vorbis, add the following entry is `sources` to copy over the config.* files so you can build on aarch64

```json
                {
                    "type": "shell",
                    "commands": [
                        "cp -p /usr/share/automake-*/config.{sub,guess} ."
                    ]
                }
```

**Comment by BeatLink** [comment]
> > Its also saying
> > ```
> > These commands were not found : mpg123, ogg123, mpg321
> >  => these file types won't be played by the 123 output : mp3, oga
> > ```
> > 
> > 
> > Thats why i installed those in the audio libs file. I thought flatpak included these
> 
> It include the libs only. :-/

Ahh all good. I readded the executables

**Comment by BeatLink** [comment]
> ```json
>                 {
>                     "type": "shell",
>                     "commands": [
>                         "cp -p /usr/share/automake-*/config.{sub,guess} ."
>                     ]
>                 }
> ```

Cool!

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
Dbus still wont work for some reason, despite the talk name. Any ideas?

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
Hmm, i keep getting this error:

`org.freedesktop.DBus.Error.AccessDenied: org.freedesktop.DBus.Error.AccessDenied`

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
It works!

IT WORKS

**Comment by BeatLink** [comment]
If there's nothing else you guys can go ahead and squash and merge the changes. I'll be uninstalling the deb build and dogfooding the flatpak test build until its uploaded to flathub

**Comment by BeatLink** [comment]
Been using it for 2 days now, works like a charm! When it is finally submitted can you give @squentin write access? Theyre the upstream developer

**Comment by nedrichards** [comment]
> nick@nedr-yoga:~$ flatpak run org.gmusicbrowser.gmusicbrowser
> neither Locale::Messages, nor Locale::gettext found -> no translations
> Creating folder /sysroot/home/nick/.var/app/org.gmusicbrowser.gmusicbrowser/config/gmusicbrowser/
> mpv version earlier than 0.7 are not supported -> mpv backend disabled
> Failed to open file “/app/share/icons/hicolor/48x48/apps/gmusicbrowser.png”: No such file or directory at /app/lib/x86_64-linux/Glib/Object/Introspection.pm line 67.

You need `copy-icon: true,` next to the icon rename.

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
bot, build org.gmusicbrowser.gmusicbrowser

**Comment by BeatLink** [comment]
For the record, that latest commit was to fix this error message

`neither Locale::Messages, nor Locale::gettext found -> no translations`

**Pull request closed without merging**